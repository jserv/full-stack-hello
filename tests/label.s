; Test for label
; Output 10! result
add $0 $1 #0    ; result
add $0 $10 #1   ; start from 10
go:
mul #0 #1 #0    ; result = result + n
sub #1 $1 #1    ; n = n - 1
jnz #1 :go      ; jump to go if #1 not zero

jmp :output       ; jump to output
add $0 $100 #1    ; should not be executed, #1 remained 0
; print the result
output:
print "10! ="
print #0        ; check result is 3628800
print #1        ; check #1 is 0
