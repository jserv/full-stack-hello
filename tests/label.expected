1+2+...+10=
55
0
